core/types/calls.cc

```diff
@@ -1858,9 +1858,11 @@ class T_must : public IntrinsicMethod {
         }
         auto ret = Types::dropNil(gs, args.args[0]->type);
         if (ret == args.args[0]->type) {
-            if (auto e = gs.beginError(args.argLoc(0), errors::Infer::InvalidCast)) {
+            auto e = args.args[0]->type.isUntyped() ? gs.beginError(args.argLoc(0), errors::Infer::MustOnUntyped)
+                                                    : gs.beginError(args.argLoc(0), errors::Infer::InvalidCast) if (e) {
                 if (args.args[0]->type.isUntyped()) {
-                    e.setHeader("`{}` called on `{}`, which is redundant", methodName, args.args[0]->type.show(gs));
+                    e.setHeader("`{}` called on `{}`, which might be redundant", methodName,
```

Comment:
Let's not change this message. Error messages in Sorbet do not second guess themselves.

Reply:
Done